modules/mcuboot/CMakeLists.txt

```diff
@@ -152,6 +152,14 @@ if(CONFIG_BOOTLOADER_MCUBOOT)
       )
   endif()
 
+  if (CONFIG_BUILD_WITH_TFM)
+    # Configure MCUBoot to be able to boot TFM
+    add_overlay_config(
+      mcuboot
+      ${ZEPHYR_NRF_MODULE_DIR}/modules/mcuboot/CONFIG_BUILD_WITH_TFM.conf
```

Comment:
Should we try to align with the names of existing overlay `.conf` files? Which are not CAPS and not prefixed with `CONIFG` ?

Reply:
No, the existing .conf files should align with CONFIG_BUILD_WITH_TFM.conf.

TODO.

If we name it like I did here we could eventually get rid of the build scripts by having child_image pick up enabled .conf files automatically. In the meantime, it will be quite clear from the filename what Kconfig option is enabling the .conf file.

Reply:
actually we should try to get rid of those automatic overlay files for child images.

The settings being enabled through those are still accessible through menuconfig and can thus be disabled again.
Or user can even do `-Dmcuboot_CONFIG_<something>=n` resulting in invalid behavior.

What we should have is a `select` / `depends on` scheme from the parent to the child, so that a child image setting may depend on a specific setting from parent, or be selected by parent and not possible to deselect.

Main question we are facing is how to handle cases where two images are compiled independently and merged or flashed later. `netcore` vs `appcore`, or `mcuboot` vs. `app`.
But that issue we already have with current approach, where the `add_overlay_config` is not applied whan a child image has been built independently of the parent image.

So for that reason, I don't think we should start adjusting other conf naming, just to change it again later.
That will just create noise twice to users familiar with current approach.

Reply:
> In the meantime, it will be quite clear from the filename what Kconfig option is enabling the .conf file.

but with this naming scheme, how would you know whether this file should be applied to mcuboot child image, netcore image, b0, or another child image ?

Reply:
> What we should have is a select / depends on scheme from the parent to the child, so that a child image setting may depend on a specific setting from parent, or be selected by parent and not possible to deselect.

That would be a great feature to have :)

> but with this naming scheme, how would you know whether this file should be applied to mcuboot child image, netcore image, b0, or another child image ?

The file's location determines which image it is applied to.

Reply:
> > but with this naming scheme, how would you know whether this file should be applied to mcuboot child image, netcore image, b0, or another child image ?
> 
> The file's location determines which image it is applied to.

but then it rightfully belongs inside that samples folder, in this case `mcuboot` sample dir itself, not the modules dir.

and what about naming when multiple conditions must be fulfilled, like:
https://github.com/nrfconnect/sdk-nrf/blob/d29f1ddec99b81a66af93b7f0882a7bd0dfa30ee/samples/CMakeLists.txt#L129
```
if (CONFIG_BT_RPMSG_NRF53)
  if (CONFIG_SOC_NRF5340_CPUAPP)
    ...
   if (CONFIG_BT_LL_SOFTDEVICE_DEFAULT)
      add_overlay_config(
        ${NETCORE_IMAGE}
        ${NRF_DIR}/subsys/bluetooth/controller/bt_ll_softdevice_hci_rpmsg.conf
        )
    ...
```

or:
https://github.com/nrfconnect/sdk-nrf/blob/d29f1ddec99b81a66af93b7f0882a7bd0dfa30ee/modules/mcuboot/CMakeLists.txt#L137-L141
where you are also depending on whether the update image is app or mcuboot linked against S1.

Reply:
> but then it rightfully belongs inside that samples folder, in this case mcuboot sample dir itself, not the modules dir.

I like it here because then we don't have to deal with multi-repo or out-of-tree patches.

I also saw that nrf/modules/mcuboot/flash_sim.overlay was here, which is similar, so I figured this was following prior art.

Also, the child_image invocation and the build scripts to overlay things onto the child image is here, so it doesn't seem unreasonable to have it here.

Reply:
> and what about naming when multiple conditions must be fulfilled, like:

This case should be covered by build scripts.

Reply:
> > but then it rightfully belongs inside that samples folder, in this case mcuboot sample dir itself, not the modules dir.
>
> I like it here because then we don't have to deal with multi-repo or out-of-tree patches.

I like it here too for the exact same reasons, but if your long term goal is:
> If we name it like I did here we could eventually get rid of the build scripts by having child_image pick up enabled .conf files automatically. 

then it should be very consistent from which folder a child image fragment gets picked up.
In sdk-nrf we have several subfolders called `mcuboot`, so just applying a fragment to a child image because it shares the name is not good choice.
Or was your intention to apply those scripts only if they are located in the folder that calls `add_child_image()` ?
That would give other challenges, for example here, where there are multiple `add_child_image()`:
https://github.com/nrfconnect/sdk-nrf/blob/master/samples/CMakeLists.txt

 > >  and what about naming when multiple conditions must be fulfilled, like:
>
> This case should be covered by build scripts.

but then how will you be able to do this:
> we could eventually get rid of the build scripts by having child_image pick up enabled .conf files automatically. 

if the file is only named after one of the conditions ?
And which condition is the *most* important condition so that it should be present in the filename ?

Until this is solved, then I prefer we have a consistent naming, where the name informs the intention of the fragment, and not when it gets included, for example:
- `secure_boot.conf` is much clearer than: `CONFIG_BOOTLOADER_MCUBOOT.conf` or `CONFIG_NRF53_UPGRADE_NETWORK_CORE.conf`
   https://github.com/nrfconnect/sdk-nrf/blob/d29f1ddec99b81a66af93b7f0882a7bd0dfa30ee/samples/CMakeLists.txt#L10
   https://github.com/nrfconnect/sdk-nrf/blob/d29f1ddec99b81a66af93b7f0882a7bd0dfa30ee/samples/CMakeLists.txt#L16
- `log_minimal.conf` is much clearer than: `CONFIG_SOC_NRF5340_CPUAPP.conf` or `CONFIG_BOOTLOADER_MCUBOOT.conf`
  https://github.com/nrfconnect/sdk-nrf/blob/d29f1ddec99b81a66af93b7f0882a7bd0dfa30ee/samples/CMakeLists.txt#L102
and there are probably more.

Until this is solved I would prefer we have a consistent naming instead of two different naming schemes.
It might be that current naming scheme is not optimal, but so far I fail to see the proposed alternative is better.

Reply:
Renamed file to tfm.conf